lab-bundle-spec.md

```diff
@@ -158,21 +153,35 @@ environment_resources:
 
 ##### GCP Project (gcp_project)
 
-attribute                     | required | type    | notes
------------------------------ | -------- | ------- | --------------------------------------
-dm_template.script            |          | path    | Relative path to a Deployment Manager directory tree.
-dm_template.custom_properties |          | array   | Array of key/value pairs.
-fleet                         |          | enum*   | Specify a Qwiklabs fleet to pull the project from.
+attribute                           | required | type    | notes
+----------------------------------- | -------- | ------- | --------------------------------------
+dm_template.start.script            |          | path    | Relative path to a Deployment Manager directory tree.
+dm_template.start.custom_properties |          | array   | Array of key/value pairs.
+dm_template.end.script              |          | path    | Relative path to a Deployment Manager directory tree.
+dm_template.end.custom_properties   |          | array   | Array of key/value pairs.
+fleet                               |          | enum*   | Specify a Qwiklabs fleet to pull the project from.
```

Comment:
variant is more or less the replacement for fleet (since resource providers are built on top of navy, not on top of the fleet model), and i believe variant should be a property on all resource types.
this does mean that labs that use weird fleets aren't compatible with the new resource model

Reply:
we can add `variant` as an optional property for all resources. can you say more about how that makes weird-fleet-labs incompatible? and is that a trade-off you're ok with?

Reply:
it's actually relying on navy that makes weird-fleet-labs incompatible, because navy doesn't support weird fleets. there's a separate thread of work on the navy side on figuring out what to do with weird fleets

Reply:
Updated